src/smc-webapp/printing.coffee

```diff
@@ -525,6 +532,9 @@ class SagewsPrinter extends Printer
                 process_collected_mesg_stdout()
                 line++
             input_lines_process(final = true)
+            # combine all html snippets to one html block
+            @_html = (h for h in @_html).join('\n')
```

Comment:
This looks suspicious to me, since for the right hand side to work, `@_html` must be an array, and after this code runs, then `@_html` is a string.  However, this is done inside of a function which is clearly called many times.   So `@_html` must be getting changed back and forth between array and string, which is just a little nerve racking to me (who grew up in statically typed C++), and feels like it might make surrounding code a little confusing to work on.  I'm sure it works fine... but I wonder if you could change it to not change the data type of `@_html`?